plugins/modules/win_domain_ou.ps1

```diff
@@ -0,0 +1,241 @@
+#!powershell
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#Requires -Module ActiveDirectory
+
+$spec = @{
+    options = @{
+        state = @{ type = "str"; choices = @("absent", "present"); default = "present" }
+        name = @{ type = "str"; required = $true }
+        protected = @{ type = "bool"; default = $false }
+        path = @{ type = "str"; required = $false }
+        filter = @{type = "str"; default = '*' }
+        recursive = @{ type = "bool"; default = $false }
+        domain_username = @{ type = "str"; }
+        domain_password = @{ type = "str"; no_log = $true }
+        domain_server = @{ type = "str" }
+        properties = @{ type = "dict" }
+    }
+    required_together = @(
+            ,@('domain_password', 'domain_username')
+        )
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$extra_args = @{}
+$onboard_extra_args = @{}
+if ($null -ne $module.Params.domain_username) {
+    $domain_password = ConvertTo-SecureString $module.Params.domain_password -AsPlainText -Force
+    $credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $module.Params.domain_username, $domain_password
+    $extra_args.Credential = $credential
+    $onboard_extra_args.Credential = $credential
+}
+if ($null -ne $module.Params.domain_server) {
+    $extra_args.Server = $module.Params.domain_server
+    $onboard_extra_args.Server = $module.Params.domain_server
+}
+if ($module.Params.properties.count -ne 0){
+    $Properties = New-Object Collections.Generic.List[string]
+    $module.Params.properties.Keys | Foreach-Object{
+        $Properties.Add($_)
+    }
+    $extra_args.Properties = $Properties
+}else{
+    $extra_args.Properties = '*'
+    $Properties = '*'
+}
+
+$extra_args.Filter = $module.Params.filter
+$check_mode = $module.CheckMode
+$name = $module.Params.name
+$protected = $module.Params.protected
+$path = $module.Params.path
+$state = $module.Params.state
+$recursive = $module.Params.recursive
+
+# setup Dynamic Params
+$params = @{}
+if ($module.Params.properties.count -ne 0){
+    $module.Params.properties.Keys | ForEach-Object{
+        $params.Add($_,$module.Params.properties.Item($_))
+    }
+}
+# capture settings to know what we are doing
+$module.Result.params = $params
+$module.Result.extra_args = $extra_args
+$module.Result.onboard_extra_args = $onboard_extra_args
+$module.Result.properties = $properties
```

Comment:
You should avoid doing this as it's a bit superfluous and could expose sensitive information. The `params` are automatically displayed in Ansible when you run with `-vvv` solving that problem and the extras are really just based on those values anyway so the data isn't really needed.

```suggestion
```